### PR TITLE
PYTHON-3829 Certificate typecast in encryption breaking build

### PR DESCRIPTION
**Pull request by NoahStapp**
Successful patch: https://spruce.mongodb.com/version/64b04375306615a3d71adc10/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: pymongo/ocsp_support.py [hunk 1]
```diff
@@ -30,6 +30,12 @@
 )
 from cryptography.hazmat.primitives.asymmetric.padding import PKCS1v15 as _PKCS1v15
 from cryptography.hazmat.primitives.asymmetric.rsa import RSAPublicKey as _RSAPublicKey
+from cryptography.hazmat.primitives.asymmetric.x448 import (
+    X448PublicKey as _X448PublicKey,
```
Comment (ShaneHarvey) [inline pymongo/ocsp_support.py:34]: This is new in cryptography 2.5 so we would need to bump our min version: https://cryptography.io/en/latest/hazmat/primitives/asymmetric/x448/#cryptography.hazmat.primitives.asymmetric.x448.X448PublicKey

We have:
```
.evergreen/test-encryption-requirements.txt:2:cryptography>=2
```

And also in pymongocrypt:
```
bindings/python/setup.py:58:        "cryptography>=2.0",
```

Comment (ShaneHarvey) [reply pymongo/ocsp_support.py:34]: We might need to add cryptography>2.5 as an explicit dependency for OCSP.

File: pymongo/ocsp_support.py [hunk 2]
```diff
@@ -132,6 +148,10 @@ def _verify_signature(
             key.verify(signature, data, algorithm)  # type: ignore[arg-type]
         elif isinstance(key, _EllipticCurvePublicKey):
             key.verify(signature, data, _ECDSA(algorithm))  # type: ignore[arg-type]
+        elif isinstance(key, _X25519PublicKey) or isinstance(
+            key, _X448PublicKey
+        ):  # Curve25519 and Curve448 keys do not require verification
```
Comment (ShaneHarvey) [inline pymongo/ocsp_support.py:153]: Neat! Can you give a reference for this?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/ocsp_support.py (hunk 2 above)
Comment (NoahStapp) [reply pymongo/ocsp_support.py:153]: For Curve448: https://crypto.stackexchange.com/questions/91710/how-do-i-validate-curve448-public-keys

For Curve25519: https://crypto.stackexchange.com/questions/87709/curve25519-key-validation and https://cr.yp.to/ecdh.html

File: pymongo/ocsp_support.py (hunk 1 above)
Comment (NoahStapp) [reply pymongo/ocsp_support.py:34]: Is bumping cryptography to >2.5 acceptable for OCSP?

Comment (ShaneHarvey) [reply pymongo/ocsp_support.py:34]: It should be fine given the version reqs here: https://github.com/pyca/pyopenssl/blob/17.2.0/setup.py#L99

Comment (ShaneHarvey) [reply pymongo/ocsp_support.py:34]: Either this or you could try/except importerror and add a fallback.

Comment (NoahStapp) [reply pymongo/ocsp_support.py:34]: So we don't need any changes for anything but the test-encryption-requirements + pymongocrypt?

Comment (ShaneHarvey) [reply pymongo/ocsp_support.py:34]: I see two options:
1) add cryptography>=2.5 as an explicit dependency for the OCSP extra. Also update test-encryption-requirements and pymongocrypt.
2) add a try/except importerror and add a fallback for this import (and maybe the other one)

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/ocsp_support.py (hunk 1 above)
Comment (NoahStapp) [reply pymongo/ocsp_support.py:34]: Option 1 seems cleaner to me, especially since 2.0 is over 5 years old.

Comment (ShaneHarvey) [reply pymongo/ocsp_support.py:34]: SGTM! could you also mention this dependency change in the changelog?

Comment (NoahStapp) [reply pymongo/ocsp_support.py:34]: Done, and PR opened for pymongocrypt change: https://github.com/mongodb/libmongocrypt/pull/668

File: pymongo/ocsp_support.py [hunk 3]
```diff
@@ -132,6 +148,10 @@ def _verify_signature(
             key.verify(signature, data, algorithm)  # type: ignore[arg-type]
         elif isinstance(key, _EllipticCurvePublicKey):
             key.verify(signature, data, _ECDSA(algorithm))  # type: ignore[arg-type]
+        elif isinstance(key, _X25519PublicKey) or isinstance(
```
Comment (ShaneHarvey) [inline pymongo/ocsp_support.py:151]: You can pass a tuple with both classes to combine these two isinstance calls into one.

**Review by ShaneHarvey — COMMENTED** [review]
Nice work! One minor nit

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**